### ISSUE TITLE
Caption-parser.js has an error in lastMatchedSample.pts

### ISSUE THREAD

**Issue by nicklive** [videojs/mux.js#223]
Hello!
I'm trying to create an hls stream with some sample file and have an error `Uncaught TypeError: Cannot read property 'pts' of undefined`

My environment:
> video.js: 7.2.2
> @videojs/http-streaming: 1.2.4
> mux.js: 4.5.1

<details>
 <summary>full log</summary>

``` 
Uncaught TypeError: Cannot read property 'pts' of undefined
    at findSeiNals (caption-parser.js:94)
    at eval (caption-parser.js:201)
    at Array.forEach (<anonymous>)
    at parseCaptionNals (caption-parser.js:183)
    at parseEmbeddedCaptions (caption-parser.js:237)
    at CaptionParser.parse (caption-parser.js:340)
    at eval (video.es.js:12759)
    at Object.eval [as callback] (video.es.js:10398)
    at cbOnce (index.js:63)
    at XMLHttpRequest.loadFunc (index.js:131)
findSeiNals @ caption-parser.js:94
(anonymous) @ caption-parser.js:201
parseCaptionNals @ caption-parser.js:183
parseEmbeddedCaptions @ caption-parser.js:237
CaptionParser.parse @ caption-parser.js:340
(anonymous) @ video.es.js:12759
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
mediaSegmentRequest @ video.es.js:12845
loadSegment_ @ video.es.js:13278
fillBuffer_ @ video.es.js:13194
monitorBufferTick_ @ video.es.js:13178
setTimeout (async)
monitorBuffer_ @ video.es.js:13173
load @ video.es.js:13095
(anonymous) @ video.es.js:14155
data.dispatcher @ video.es.js:798
trigger @ video.es.js:840
EventTarget.trigger @ video.es.js:1062
haveMetadata @ video.es.js:10143
(anonymous) @ video.es.js:10135
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
(anonymous) @ video.es.js:10134
data.dispatcher @ video.es.js:798
trigger @ video.es.js:840
EventTarget.trigger @ video.es.js:1062
(anonymous) @ video.es.js:10143
setTimeout (async)
haveMetadata @ video.es.js:10143
(anonymous) @ video.es.js:10135
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
(anonymous) @ video.es.js:10134
data.dispatcher @ video.es.js:798
trigger @ video.es.js:840
EventTarget.trigger @ video.es.js:1062
(anonymous) @ video.es.js:10143
setTimeout (async)
haveMetadata @ video.es.js:10143
(anonymous) @ video.es.js:10135
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
(anonymous) @ video.es.js:10134
data.dispatcher @ video.es.js:798
trigger @ video.es.js:840
EventTarget.trigger @ video.es.js:1062
(anonymous) @ video.es.js:10143
setTimeout (async)
haveMetadata @ video.es.js:10143
(anonymous) @ video.es.js:10135
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
(anonymous) @ video.es.js:10134
data.dispatcher @ video.es.js:798
trigger @ video.es.js:840
EventTarget.trigger @ video.es.js:1062
(anonymous) @ video.es.js:10143
setTimeout (async)
haveMetadata @ video.es.js:10143
(anonymous) @ video.es.js:10169
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
media @ video.es.js:10168
(anonymous) @ video.es.js:14148
data.dispatcher @ video.es.js:798
trigger @ video.es.js:840
EventTarget.trigger @ video.es.js:1062
(anonymous) @ video.es.js:10187
(anonymous) @ video.es.js:10398
cbOnce @ index.js:63
loadFunc @ index.js:131
load (async)
_createXHR @ index.js:172
createXHR @ index.js:51
XhrFunction @ video.es.js:10392
start @ video.es.js:10183
load @ video.es.js:10180
MasterPlaylistController @ video.es.js:14137
src @ video.es.js:14630
handleSource @ video.es.js:14675
_Tech.setSource @ video.es.js:3806
(anonymous) @ video.es.js:8354
ready @ video.es.js:1613
techCall_ @ video.es.js:8354
(anonymous) @ video.es.js:8728
ready @ video.es.js:1613
src_ @ video.es.js:8724
(anonymous) @ video.es.js:8706
setSourceHelper @ video.es.js:3845
(anonymous) @ video.es.js:3845
setSource @ video.es.js:14554
setSourceHelper @ video.es.js:3840
setSourceHelper @ video.es.js:3845
(anonymous) @ video.es.js:3820
bound @ video.es.js:928
(anonymous) @ video.es.js:1958
setTimeout (async)
setTimeout @ video.es.js:1958
setSource @ video.es.js:3820
src @ video.es.js:8704
set src @ index.js:61
VideoJSPlayerAdapter @ index.js:52
PlayerAdapter @ PlayerAdapter.js:80
initPlayer @ PlayerProvider.js:149
componentWillUpdate @ PlayerProvider.js:170
renderComponent @ preact.mjs:515
rerender @ preact.mjs:92
Promise.then (async)
enqueueRender @ preact.mjs:83
setState @ preact.mjs:698
PlayerProvider.onVideoMount.ref @ PlayerProvider.js:62
componentDidMount @ Video.js:8
flushMounts @ preact.mjs:202
diff @ preact.mjs:220
render @ preact.mjs:708
render @ index.js:30
PlayerUI @ index.js:26
mount @ index.js?c1d2:63
componentDidMount @ index.js?c1d2:30
(anonymous) @ ReactCompositeComponent.js?059b:264
measureLifeCyclePerf @ ReactCompositeComponent.js?059b:75
(anonymous) @ ReactCompositeComponent.js?059b:263
notifyAll @ CallbackQueue.js?0acc:76
close @ ReactReconcileTransaction.js?bf13:80
closeAll @ Transaction.js?0391:209
perform @ Transaction.js?0391:156
perform @ Transaction.js?0391:143
perform @ ReactUpdates.js?3983:89
flushBatchedUpdates @ ReactUpdates.js?3983:172
closeAll @ Transaction.js?0391:209
perform @ Transaction.js?0391:156
batchedUpdates @ ReactDefaultBatchingStrategy.js?ed0e:62
enqueueUpdate @ ReactUpdates.js?3983:200
enqueueUpdate @ ReactUpdateQueue.js?cd58:24
enqueueSetState @ ReactUpdateQueue.js?cd58:218
ReactComponent.setState @ ReactBaseClasses.js?8885:64
Overview._this.onChange @ Overview.js?6922:49
EventEmitter.emit @ events.js?faa1:78
emitChange @ ChannelStatusStore.js?e482:97
loadData @ ChannelStatusStore.js?e482:43
(anonymous) @ OverviewActions.js?6ff8:33
j @ jquery.min.js:2
fireWith @ jquery.min.js:2
x @ jquery.min.js:4
b @ jquery.min.js:4
XMLHttpRequest.send (async)
send @ jquery.min.js:4
ajax @ jquery.min.js:4
loadChannelData @ OverviewActions.js?6ff8:29
componentDidMount @ Overview.js?6922:36
(anonymous) @ ReactCompositeComponent.js?059b:264
measureLifeCyclePerf @ ReactCompositeComponent.js?059b:75
(anonymous) @ ReactCompositeComponent.js?059b:263
notifyAll @ CallbackQueue.js?0acc:76
close @ ReactReconcileTransaction.js?bf13:80
closeAll @ Transaction.js?0391:209
perform @ Transaction.js?0391:156
batchedMountComponentIntoNode @ ReactMount.js?25ea:126
perform @ Transaction.js?0391:143
batchedUpdates @ ReactDefaultBatchingStrategy.js?ed0e:62
batchedUpdates @ ReactUpdates.js?3983:97
_renderNewRootComponent @ ReactMount.js?25ea:319
_renderSubtreeIntoContainer @ ReactMount.js?25ea:401
render @ ReactMount.js?25ea:422
_callee$ @ app.js?4560:127
tryCatch @ runtime.js?f3a2:65
invoke @ runtime.js?f3a2:303
prototype.(anonymous function) @ runtime.js?f3a2:117
step @ app.js?4560:1
(anonymous) @ app.js?4560:1
Promise.then (async)
step @ app.js?4560:1
(anonymous) @ app.js?4560:1
Promise.then (async)
step @ app.js?4560:1
(anonymous) @ app.js?4560:1
Promise.then (async)
step @ app.js?4560:1
(anonymous) @ app.js?4560:1
Promise.then (async)
step @ app.js?4560:1
(anonymous) @ app.js?4560:1
Promise.then (async)
step @ app.js?4560:1
(anonymous) @ app.js?4560:1
(anonymous) @ app.js?4560:1
init @ app.js?4560:26
(anonymous) @ app.js?4560:139
./js/app.js @ browser-bundle.min.js:603
__webpack_require__ @ browser-bundle.min.js:79
0 @ browser-bundle.min.js:9234
__webpack_require__ @ browser-bundle.min.js:79
checkDeferredModules @ browser-bundle.min.js:46
(anonymous) @ browser-bundle.min.js:152
(anonymous) @ browser-bundle.min.js:155
```

</details>

I've done some research for this bug and found that here in `caption-parser.js` variable `lastMatchedSample` have no initial value, and later, if `matchingSample` is null `lastMatchedSample.pts` can't be reached, because it is undefined
https://github.com/videojs/mux.js/blob/1c888315c09148d6c9c2e716d2c8099f1c3e93d9/lib/mp4/caption-parser.js#L64-L97

Please, can you clarify what this error about and can this error influence an video playback in video.js (I have some video freezes after that error)?

**Comment by nicklive** [comment]
I've tried to use older version of `video.js` (6.12.1) (with old version of mux.js) and this problem is not revealed.

**Comment by hikingit** [comment]
Was getting the same error on certain streams. Had to back to version 7.1.0, because this error occurs on newer versions.

**Comment by mgara** [comment]
I had the same issue today .. I managed to 'work around' it  by disabling the EIA-608 from my h264 stream
https://stackoverflow.com/questions/48177694/removing-eia-608-closed-captions-from-h-264-without-reencode
Please let me know if you need more info
I'm using HLS with mp4ts

**Comment by gkatsev** [comment]
So, this should be addressed or at least worked around by #321. In the original PR #235, there's been some discussion that this could either be a content issue or a more fundamental issue in our caption parsing code (https://github.com/videojs/mux.js/pull/235#issuecomment-478707570). We'd need to investigate it a bit more thoroughly. There is sample content in this comment https://github.com/videojs/mux.js/pull/235#issuecomment-478041033.
